cli/cli-utils/src/packageIsInstallable.ts

```diff
@@ -4,6 +4,17 @@ import { logger, globalWarn } from '@pnpm/logger'
 import { checkPackage, UnsupportedEngineError, type WantedEngine } from '@pnpm/package-is-installable'
 import { type SupportedArchitectures } from '@pnpm/types'
 
+export function parsePackageManager (packageManager: string): { name: string, version: string | undefined } {
+  const [name, pmReference] = packageManager.split('@')
+  // pmReference is semantic versioning, not URL
+  if (pmReference.includes(':')) return { name, version: undefined }
+  const [version] = pmReference.split('+')
```

Comment:
This omits the integrity; is supporting that possible? Or is the corepack blob different than what pnpm is downloading?

Reply:
What is that blob about? We could support an sha512 integrity checksum. The same that we use for dependencies in the lockfile. But that would involve a bigger rewrite. If it is required, we can work on it in a separate PR. Also, we will only support installing pnpm from the registry.

Reply:
I can't recall if corepack downloads the npm package or something else, but it looks like pnpm's npm package is self-contained so likely it was downloading that? (I have not checked)

Reply:
yes, it must be downloading npm from the registry.

Reply:
Regarding `pmReference.includes(':')`

I get errors since `pmReference` is undefined in certain cases. It's kind of hard to nail down...